stat/stat.go

```diff
@@ -14,12 +14,15 @@ import (
 // CumulantKind specifies the behavior for calculating the empirical CDF or Quantile
 type CumulantKind int
 
+// List of supported CumulantKind values for the Quantile function.
+// Constant values should match the R nomenclature. See
+// https://en.wikipedia.org/wiki/Quantile#Estimating_the_quantiles_of_a_population
 const (
-	// Constant values should match the R nomenclature. See
-	// https://en.wikipedia.org/wiki/Quantile#Estimating_the_quantiles_of_a_population
-
 	// Empirical treats the distribution as the actual empirical distribution.
 	Empirical CumulantKind = 1
+
+	// AvgDiscont is like Empirical but with averaging at discontinuities.
```

Comment:
type 2 is something people coming from `R` might expect.
also, it's the expected behaviour (at least for me) when dealing with unique, non-weighted data.

I'd leave it there and volounteer myself to implement type 4 (in the next PR)

Reply:
We can have it for compatibility, but I really don't think it's what most people want in general. I agree that it gives the answer you're expecting for unique, non-weighted data, but Type 4 gives the same answer. I think we should encourage people to use Type 4 in documentation and examples, because it gives the same behavior as type 2 in the common case, and gives a much less surprising answer in all other cases (p != 0.5 or weights != nil).

Reply:
I think the name should be something like `HalfBetween`, because it doesn't really average across the discontinuity in a meaningful sense, it just replaces the discontinuity with a different discontinuity. Instead, it takes half of the sample ends if the quantile falls between samples. 

With weighted data, it's not clear to me if the extension is half, or if the extension is the average of the two weights (i.e. (w_1 x_ + w_2 x_2)/(w_1 + w_2)). I think the half probably makes more sense, because that's how it would work if you replicated the data.